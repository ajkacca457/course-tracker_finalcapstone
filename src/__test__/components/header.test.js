import React from 'react';
import TestRenderer from 'react-test-renderer';
import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom/extend-expect';
import Header from '../../components/header';
import store from '../../redux/store';

describe('Home Component', () => {
  it('renders correctly', () => {
    const hometree = TestRenderer
      .create(
        <Provider store={store}>
          <BrowserRouter>
            <Header />
          </BrowserRouter>
        </Provider>,
      )
      .toJSON();
    expect(hometree).toMatchSnapshot();
  });

  test('renders the title text of the Home page', () => {
    const { getByText } = render(
      <Provider store={store}>
        <BrowserRouter>
          <Header />
        </BrowserRouter>
      </Provider>,
    );
    const title = getByText('Course Tracker');
    expect(title).toBeInTheDocument();
  });
});
