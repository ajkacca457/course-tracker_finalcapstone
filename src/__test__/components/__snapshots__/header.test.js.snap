// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component renders correctly 1`] = `
<div
  className="header bg-dark px-2 my-0 d-flex justify-content-between align-items-center"
>
  <h4
    className="text-center text-white pt-1 mb-3"
  >
    <i
      className="fas fa-chalkboard-teacher"
    />
     
    Course Tracker
  </h4>
  <button
    className="d-none"
    onClick={[Function]}
    type="button"
  >
     
    <i
      className="fas fa-power-off text-white"
    />
  </button>
</div>
`;
